numpyro/util.py

```diff
@@ -298,3 +298,42 @@ def unravel_pytree(arr):
         return tree_unflatten(treedef, unravel_list(arr))
 
     return flat, unravel_pytree
+
+
+def soft_vmap(fn, xs, batch_ndims=1, chunk_size=None):
+    """
+    Vectorizing map a function `fn` over `batch_ndims` leading axes of `xs`.
+    The job is splitted into chunks to reduce memory usages.
```

Comment:
nits: 
Vectorizing map that maps a function `fn` ..
splitted --> split

Maybe we can replace the last line with: This uses `jax.vmap` over smaller chunks of the batch dimension to keep memory usage constant.

Reply:
Thanks for suggesting!